File: test/classi_struct.cpp
class personclass{
private:
  float __height;

public:
  int age;

  personclass() 
  :__height(0.0),age(0) {}

personclass(int age, float height) 
  {
    this->__height = height;
    this->age = age;
  }

};
struct personstruct{
  int age;

  float height;

  personclass person;

  personstruct() 
  :age(0),height(0.0),person() {}

personstruct(int age, float height, personclass person) 
  {
    this->age = age;
    this->height = height;
    this->person = person;
  }

};
struct window_functor{
  void __call__(personclass personS) 
  {
    personS.age += 1;
  }

};

File: test/classi_struct.py
@transpile
class personclass:
    age:int
    __height:float
    __weight:float=10
    def __init__(self, age:int, height:float):
        self.__height = height
        self.age = age



class personstruct:
    age:int
    height:float

    @transpile
    def __init__(self, age:int, height:float, person:personclass):
        self.age = age
        self.height = height
        self.person = person
    def reset(self,person:personclass):
        self.age = 0
        self.height = 0.0
        self.personC = person
@transpile
class window_functor:
    def __call__(self,personS:personclass):
        personS.age+=1